lua/lualine.lua

```diff
@@ -64,8 +64,8 @@ local function StatusLine(isFocused)
   end
   if type(M.theme) == 'string' then
     M.theme = utils.setTheme(M.theme)
-    highlight.createHighlightGroups(M.theme)
   end
+  highlight.createHighlightGroups(M.theme)
```

Comment:
This is a really cheap fix to this problem. This `M.Statusline` runs every time a change happens in the statusline. Your solution makes lualine run all highlight commands run on every change. A better solution might be to copy the whole if block to the `M.status()` function, see if that resolves this issue.

Reply:
Tried that, but the highlights will get called before the sections are drawn. Though, M.Statusline only appears to be called on Win/Buf Enter/Leave, so it's doesn't seem overkill to have the highlights applied then.

Reply:
Please check if https://github.com/hoob3rt/lualine.nvim/tree/theme-highlight-fix resolves this issue.

Reply:
This still would not resolve things if the theme is not a module in this repo.  If a third party theme is passed, it would not be highlighted.

Reply:
Oh you are right, but I think a better solution is needed than just spamming a highlight all the, let me know if you come up with something, I will try to tackle this tomorrow.